pkg/controller/populators/populator-base.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2023 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package populators
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	cc "kubevirt.io/containerized-data-importer/pkg/controller/common"
+	featuregates "kubevirt.io/containerized-data-importer/pkg/feature-gates"
+	"kubevirt.io/containerized-data-importer/pkg/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type pvcModifierFunc func(pvc *corev1.PersistentVolumeClaim, source client.Object)
+
+// ReconcilerBase members
+type ReconcilerBase struct {
+	client          client.Client
+	recorder        record.EventRecorder
+	scheme          *runtime.Scheme
+	log             logr.Logger
+	featureGates    featuregates.FeatureGates
+	sourceKind      string
+	installerLabels map[string]string
+}
+
+const (
+	dataSourceRefField = "spec.dataSourceRef"
+)
+
+// Interface to store populator-specific methods
+type populatorController interface {
+	// Returns the specific populator CR
+	getPopulationSource(namespace, name string) (client.Object, error)
+	// Returns PVC' annotations according to the population source
+	updateAnnotations(pvc *corev1.PersistentVolumeClaim, source client.Object)
```

Comment:
the callee can technically do anything with `pvc`, right?  So why call the function `updateAnnotations`?

Reply:
This was to follow what we already do on the DV controller, but maybe something like `preparePVCForPopulation` or `setupPVCForPopulation` would be more descriptive.

Reply:
Or just `updatePVC` I don't really care

Reply:
Updated it (ended up using `preparePVCForPopulation` but I can change it if preferred).